### PR TITLE
ISPN-8651 Use functional commands for JCache operations

### PR DESCRIPTION
**Pull request by rvansa**
https://issues.jboss.org/browse/ISPN-8651
https://issues.jboss.org/browse/ISPN-8676 (partially, do not close the JIRA)

### CODE REVIEW

File: jcache/embedded/src/main/java/org/infinispan/jcache/embedded/marshall/SingletonExternalizer.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.commons.util.Util;
+
+public class SingletonExternalizer<T> implements AdvancedExternalizer<T> {
```
Comment (galderz) [inline jcache/embedded/src/main/java/org/infinispan/jcache/embedded/marshall/SingletonExternalizer.java]: Could this (and `SuppliedExternalizer`) be useful for other externalizers in other modules? If so we could have these class in core instead? WDYT?

File: jcache/embedded/src/test/java/org/infinispan/jcache/JCacheTwoCachesBasicOpsTest.java [hunk 2]
```diff
@@ -16,7 +16,7 @@
  */
 @Test(testName = "org.infinispan.jcache.JCacheTwoCachesBasicOpsTest", groups = "functional")
 @CleanupAfterMethod
-public class JCacheTwoCachesBasicOpsTest extends AbstractTwoCachesBasicOpsTest {
+   public class JCacheTwoCachesBasicOpsTest extends AbstractTwoCachesBasicOpsTest {
```
Comment (galderz) [inline jcache/embedded/src/test/java/org/infinispan/jcache/JCacheTwoCachesBasicOpsTest.java]: What's this for?

**Comment by rvansa** [comment]
@galderz Moved the generic externalizers to commons and removed the whitespace (typo).

**Comment by galderz** [comment]
Build errors:

```
[ERROR] /home/infinispan/workspace/Infinispan_PR-5670-L74FREARJGVWJFT3K3SN5UGJEWAJ22W7XCQRYSSUJLD2L7UEVAVA/commons/src/main/java/org/infinispan/commons/marshall/SingletonExternalizer.java:8:1: Redundant import from the same package - org.infinispan.commons.marshall.AdvancedExternalizer. [RedundantImport]
[ERROR] /home/infinispan/workspace/Infinispan_PR-5670-L74FREARJGVWJFT3K3SN5UGJEWAJ22W7XCQRYSSUJLD2L7UEVAVA/commons/src/main/java/org/infinispan/commons/marshall/SuppliedExternalizer.java:9:1: Redundant import from the same package - org.infinispan.commons.marshall.AdvancedExternalizer. [RedundantImport]
```

**Comment by galderz** [comment]
290 failures

**Comment by rvansa** [comment]
Damned uberjars. Had to add JCache test-depedency to `infinispan-embedded-query-it` and `:integrationtests-osgi`

**Comment by rvansa** [comment]
Actually the last fix isn't a proper fix... still on ice.

**Comment by rvansa** [comment]
Included better fix.

**Comment by rvansa** [comment]
The JCache test failure is ISPN-8710, fixed in master.

**Comment by gustavonalle** [comment]
merged, thanks @rvansa

**Comment by gustavonalle** [comment]
Ops...last commit had compilation failures after rebasing, fixing...

**Pull request merged**